src/backends/r1cs/snarkjs.rs

```diff
@@ -357,93 +294,3 @@ impl WitnessWriter {
-        let mut r1cs = R1CS::new();
-
-        // mock a constraint
-        let span = Span::default();
-        let var1_val = 2;
-        let public_input_val = 3;
-        let res_val = var1_val * public_input_val;
-        let var1 = r1cs.new_internal_var(Value::Constant(R1csBls12381Field::from(var1_val)), span);
-        let public_input_var = r1cs.add_public_input(
-            Value::Constant(R1csBls12381Field::from(public_input_val)),
-            span,
-        );
-        let sum_var = r1cs.mul(&var1, &public_input_var, span);
-        r1cs.add_public_output(Value::PublicOutput(Some(sum_var)), span);
-        let public_output_val = res_val;
-
-        // convert witness to snarkjs format
-        let mut snarkjs_exporter = super::SnarkjsExporter::new(r1cs);
-        assert_eq!(
-            snarkjs_exporter.witness_map.len(),
-            snarkjs_exporter.r1cs_backend.witness_vars.len()
-        );
-
-        // mock finalizing the circuit
-        snarkjs_exporter.r1cs_backend.finalized = true;
-        let mut witness_env = WitnessEnv::default();
-        let generated_witness = snarkjs_exporter
-            .r1cs_backend
-            .generate_witness(&mut witness_env);
-
-        let rearranged_witness = snarkjs_exporter.restructure_witness(generated_witness.unwrap());
-
-        assert_eq!(
-            rearranged_witness,
-            vec![
-                // first var is always 1
-                BigInt::from(1),
-                // output ordered before input
-                BigInt::from(public_output_val),
-                // public input
-                BigInt::from(public_input_val),
-                // private inputs (the rest of the witness vars)
-                BigInt::from(var1_val),
-                BigInt::from(res_val),
-            ]
-        );
-
-        let restructure_constraints = snarkjs_exporter.restructure_constraints();
-        assert_ne!(restructure_constraints.len(), 0);
-
-        for (rc, oc) in restructure_constraints
-            .iter()
-            .zip(snarkjs_exporter.r1cs_backend.constraints.iter())
-        {
-            // concat the terms from a b c
-            let all_original_terms = [oc.a.terms.clone(), oc.b.terms.clone(), oc.c.terms.clone()];
-            let all_restructured_terms =
-                [rc.a.terms.clone(), rc.b.terms.clone(), rc.c.terms.clone()];
-
-            for (oterms, rterms) in all_original_terms.iter().zip(all_restructured_terms) {
-                assert_eq!(oterms.len(), rterms.len());
-                for original_var in oterms.keys() {
-                    // check if the original var is in the restructured terms after reordering
-                    let new_index = snarkjs_exporter
-                        .witness_map
-                        .get(&original_var.index)
-                        .unwrap();
-                    assert!(rterms.contains_key(new_index));
-                }
-            }
-        }
-    }
-}
```

Comment:
nice to see how much code we're removing through this PR :D

Reply:
yeah, it is indeed simplify a lot things. Thanks for the suggestions!